tests/src/test/scala/org/http4s/util/DecodeSpec.scala

```diff
@@ -55,5 +55,11 @@ class DecodeSpec extends Http4sSpec {
         decoded must_== expected
       }
     }
+
+    "drop Byte Order Mark" in {
```

Comment:
Probably exceedingly rare, but what if the BOM spans chunks?

Reply:
Given, that we exclude cases of it appearing in the middle of the stream, I don't think it would ever happen as we use `unconsN`. It might be more likely in fs2 implementation, where they use `uncons1`.

Reply:
Having written this it became clear to me that it is actually unnecessary to check `chunk.size >= 3`.
I took it from fs2, but because of `unconsN` it is redundant in our implementation.